apps/spruce/src/pages/task/metadata/TaskOwnership/TaskOwnership.test.tsx

```diff
@@ -0,0 +1,223 @@
+} from "gql/generated/types";
+import { TASK_OWNER_TEAM } from "gql/queries";
+import TaskOwnership from ".";
+
+const taskId = "task123";
+const execution = 5;
+
+const taskOwnerTeamMock: ApolloMock<
+  TaskOwnerTeamsForTaskQuery,
+  TaskOwnerTeamsForTaskQueryVariables
+> = {
+  request: {
+    query: TASK_OWNER_TEAM,
+    variables: { taskId, execution },
+  },
+  result: {
+    data: {
+      task: {
+        __typename: "Task",
+        id: taskId,
+        execution,
+        taskOwnerTeam: {
+          __typename: "TaskOwnerTeam",
+          assignmentType: "auto",
+          jiraProject: "EVG",
+          messages: "Assigned based on task history",
+          teamName: "Platform Team",
+        },
+      },
+    },
+  },
+};
+
+const taskOwnerTeamEmptyMock: ApolloMock<
+  TaskOwnerTeamsForTaskQuery,
+  TaskOwnerTeamsForTaskQueryVariables
+> = {
+  request: {
+    query: TASK_OWNER_TEAM,
+    variables: { taskId, execution },
+  },
+  result: {
+    data: {
+      task: {
+        __typename: "Task",
+        id: taskId,
+        execution,
+        taskOwnerTeam: null,
+      },
+    },
+  },
+};
+
+const taskOwnerTeamNoNameMock: ApolloMock<
+  TaskOwnerTeamsForTaskQuery,
+  TaskOwnerTeamsForTaskQueryVariables
+> = {
+  request: {
+    query: TASK_OWNER_TEAM,
+    variables: { taskId, execution },
+  },
+  result: {
+    data: {
+      task: {
+        __typename: "Task",
+        id: taskId,
+        execution,
+        taskOwnerTeam: {
+          __typename: "TaskOwnerTeam",
+          assignmentType: "auto",
+          jiraProject: "EVG",
+          messages: "Assigned based on task history",
+          teamName: "",
+        },
+      },
+    },
+  },
+};
+
+const loadingMock: ApolloMock<
+  TaskOwnerTeamsForTaskQuery,
+  TaskOwnerTeamsForTaskQueryVariables
+> = {
+  request: {
+    query: TASK_OWNER_TEAM,
+    variables: { taskId, execution },
+  },
+  result: { data: undefined },
+};
+
+describe("TaskOwnership", () => {
+  it("renders nothing while loading", async () => {
+    render(
+      <MockedProvider mocks={[loadingMock]}>
+        <TaskOwnership execution={execution} taskId={taskId} />
+      </MockedProvider>,
+    );
+
+    // Should not find the component while loading
+    expect(
+      screen.queryByDataCy("task-metadata-task-ownership"),
+    ).not.toBeInTheDocument();
+  });
+
+  it("renders team name when available", async () => {
+    render(
+      <MockedProvider mocks={[taskOwnerTeamMock]}>
+        <TaskOwnership execution={execution} taskId={taskId} />
+      </MockedProvider>,
+    );
+
+    await waitFor(() => {
+      expect(
+        screen.getByDataCy("task-metadata-task-ownership"),
+      ).toBeInTheDocument();
+    });
+
+    expect(screen.getByText("Task Owner:")).toBeInTheDocument();
+    expect(screen.getByText("Platform Team")).toBeInTheDocument();
+
+    // Hover over the tooltip
+    act(() => {
+      screen.getByDataTestid("info-sprinkle-icon").focus();
+    });
```

Comment:
We should be using `userEvent` library instead to test hover
Not sure if you'll still need `act` if you're going to be `await`ing the `hover` event